src/Controls/src/Core/PointerGestureRecognizer.cs

```diff
@@ -0,0 +1,101 @@
+
+		public static readonly BindableProperty PointerMovedCommandProperty = BindableProperty.Create(nameof(PointerMovedCommand), typeof(ICommand), typeof(PointerGestureRecognizer), null);
+
+		public static readonly BindableProperty PointerMovedCommandParameterProperty = BindableProperty.Create(nameof(PointerMovedCommandParameter), typeof(object), typeof(PointerGestureRecognizer), null);
+
+		public PointerGestureRecognizer()
+		{
+		}
+
+		public event EventHandler<PointerEventArgs>? PointerEntered;
+		public event EventHandler<PointerEventArgs>? PointerExited;
+		public event EventHandler<PointerEventArgs>? PointerMoved;
+
+		public ICommand PointerEnteredCommand
+		{
+			get { return (ICommand)GetValue(PointerEnteredCommandProperty); }
+			set { SetValue(PointerEnteredCommandProperty, value); }
+		}
+
+		public ICommand PointerEnteredCommandParameter
+		{
+			get { return (ICommand)GetValue(PointerEnteredCommandParameterProperty); }
+			set { SetValue(PointerEnteredCommandParameterProperty, value); }
+		}
+		public ICommand PointerExitedCommand
+		{
+			get { return (ICommand)GetValue(PointerExitedCommandProperty); }
+			set { SetValue(PointerExitedCommandProperty, value); }
+		}
+		public ICommand PointerExitedCommandParameter
+		{
+			get { return (ICommand)GetValue(PointerExitedCommandParameterProperty); }
+			set { SetValue(PointerExitedCommandParameterProperty, value); }
+		}
+
+		public ICommand PointerMovedCommand
+		{
+			get { return (ICommand)GetValue(PointerMovedCommandProperty); }
+			set { SetValue(PointerMovedCommandProperty, value); }
+		}
+
+		public ICommand PointerMovedCommandParameter
+		{
+			get { return (ICommand)GetValue(PointerMovedCommandParameterProperty); }
+			set { SetValue(PointerMovedCommandParameterProperty, value); }
+		}
+		
+		internal void SendPointerEntered(View sender, Func<IElement?, Point?>? getPosition)
+		{
+			ICommand cmd = PointerEnteredCommand;
+			if (cmd != null && cmd.CanExecute(PointerEnteredCommandParameter))
+				cmd.Execute(PointerEnteredCommandParameter);
+
+			EventHandler<PointerEventArgs>? handler = PointerEntered;
+			if (handler != null)
+				handler?.Invoke(sender, new PointerEventArgs(getPosition));
+		}
+
+		internal void SendPointerExited(View sender, Func<IElement?, Point?>? getPosition)
+		{
+			ICommand cmd = PointerExitedCommand;
+			if (cmd != null && cmd.CanExecute(PointerExitedCommandParameter))
+				cmd.Execute(PointerExitedCommandParameter);
+
+			EventHandler<PointerEventArgs>? handler = PointerExited;
+			if (handler != null)
+				handler?.Invoke(sender, new PointerEventArgs(getPosition));
+		}
+
+		internal void SendPointerMoved(View sender, Func<IElement?, Point?>? getPosition)
+		{
+			ICommand cmd = PointerMovedCommand;
+			if (cmd != null && cmd.CanExecute(PointerMovedCommandParameter))
```

Comment:
```suggestion
			if (cmd?.CanExecute(PointerMovedCommandParameter))
```

Reply:
Intellisense tells me this won't work since it can't implicitly convert type bool? to bool, so I'll leave this one as is for now.
Alternatively, I can cast it as a bool, but I prefer to leave it as is. Let me know what you think!

Reply:
you can do it like this

```C#
cmd?.CanExecute(PointerMovedCommandParameter) == true
```

Reply:
ahh of course, fixed!